src/schedule/kat_key_schedule.rs

```diff
@@ -209,14 +215,16 @@ fn run_test_vectors() {
             }
 
             let joiner_secret = JoinerSecret::new(ciphersuite, &commit_secret, &init_secret);
-            assert_eq!(hex_to_bytes(&epoch.joiner_secret), joiner_secret.as_slice());
+            if hex_to_bytes(&epoch.joiner_secret) != joiner_secret.as_slice() {
+                return Err(KSTestVectorError::JoinerSecretMismatch);
```

Comment:
This is a bad idea. This code is supposed to test if things are working and hence should assert at the exact point where something goes wrong. Why do you think you need errors?

Reply:
In the context of interop, I'd like to call this function and, if it fails, it would be nice to get some information back, which can then be reported to the test runner. If I have asserts here, I could still try to use `catch_unwind`, but using errors seemed to be the cleaner approach. I'm happy to use another approach here if you have a suggestion.

In case it helps: The calling function should panic right after an error is returned, so not too far from where things go wrong.

Reply:
If you really want to return errors here, which might make sense in the case you describe, you should still keep the assert but only in `cfg(test)` to keep the test as it is.

Reply:
Aah, that's a great idea! I hadn't thought about that.